App/AuthProvider.js

```diff
@@ -3,223 +3,119 @@
-      this.customFileProtocolName
-    );
-    const codePromise = authCodeListener.start();
-    authWindow.loadURL(navigateUrl);
-    const code = await codePromise;
-    authCodeListener.close();
-    return code;
-  }
-
-  /**
-   * Handles the response from a popup or redirect. If response is null, will check if we have any accounts and attempt to sign in.
-   * @param response
-   */
-  async handleResponse(response) {
-    if (response !== null) {
-      this.account = response.account;
-    } else {
-      this.account = await this.getAccount();
+
+    async getTokenSilent(tokenRequest) {
+        try {
+            return await this.clientApplication.acquireTokenSilent(tokenRequest);
+        } catch (error) {
+            console.log('Silent token acquisition failed, acquiring token interactive');
+            return await this.clientApplication.acquireTokenInteractive({
+                ...tokenRequest,
+                successTemplate: 'Acquired token interactive.',
+                openBrowser: this.openBrowser,
+            });
+        }
     }
 
-    return this.account;
-  }
-
-  /**
-   * Calls getAllAccounts and determines the correct account to sign into, currently defaults to first account found in cache.
-   * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/docs/Accounts.md
-   */
-  async getAccount() {
-    // need to call getAccount here?
-    const cache = this.clientApplication.getTokenCache();
-    const currentAccounts = await cache.getAllAccounts();
-
-    if (currentAccounts === null) {
-      console.log("No accounts detected");
-      return null;
+    /**
+     * Handles the response from a popup or redirect. If response is null, will check if we have any accounts and attempt to sign in.
+     * @param response
+     */
+    async handleResponse(response) {
+        if (response !== null) {
+            this.account = response.account;
+        } else {
+            this.account = await this.getAccount();
+        }
+
+        return this.account;
     }
 
-    if (currentAccounts.length > 1) {
-      // Add choose account code here
-      console.log(
-        "Multiple accounts detected, need to add choose account code."
-      );
-      return currentAccounts[0];
-    } else if (currentAccounts.length === 1) {
-      return currentAccounts[0];
-    } else {
-      return null;
+    /**
+     * Calls getAllAccounts and determines the correct account to sign into, currently defaults to first account found in cache.
+     * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/docs/Accounts.md
+     */
+    async getAccount() {
+        const cache = this.clientApplication.getTokenCache();
+        const currentAccounts = await cache.getAllAccounts();
+
+        if (currentAccounts === null) {
+            console.log('No accounts detected');
+            return null;
+        }
+
```

Comment:
We're _absolutely sure_ that `await cache.getAllAccounts();` will return `null` when there are no accounts?

There is some potential dreaded chicanery `===`

![image](https://user-images.githubusercontent.com/93682173/196508284-a57b2791-5375-4737-8e43-3c6ae66f402f.png)

Reply:
`!currentAccounts` might be better.